include/ccf/endpoint_registry.h

```diff
@@ -170,6 +173,30 @@ namespace ccf::endpoints
       const ReadOnlyEndpointFunction& f,
       const AuthnPolicies& ap);
 
+    /** Create a new endpoint with a commit handler.
+     *
+     * Like make_endpoint but takes a functor to call once the transaction has
+     * been committed, but before consensus has completed.
+     *
+     * @param l Functor which will be invoked after the transaction has been
+     * assigned an ID.
+     */
+    virtual Endpoint make_endpoint_with_commit_handler(
```

Comment:
```suggestion
    virtual Endpoint make_endpoint_with_local_commit_handler(
```

I think these methods should all include `local_commit` in the name, rather than `commit`, to avoid confusion. One day we may find better terminology for local commit, as we've largely erased it from the public API, but I don't think we have a better alternative name here.

Reply:
That does make it clearer, wasn't sure on the status of erasing the local/global from the API